### ISSUE TITLE
"ansible_service_mgr" is not correctly detected in ubuntu 14.04

### ISSUE THREAD

**Issue by pescobar** [ansible/ansible#32001]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

setup module 

##### ANSIBLE VERSION
```
root@test1:~# ansible --version
ansible 2.4.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/vagrant/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.6 (default, Mar 22 2014, 22:59:56) [GCC 4.8.2]
```

##### CONFIGURATION

```
root@test1:~# ansible-config dump --only-changed
root@test1:~#
```

##### OS / ENVIRONMENT
```
root@test1:~# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty
```
##### SUMMARY

Using ansible 2.4.0.0 to administer a ubuntu14.04 machine `ansible_service_mgr` is not correctly detected. 

##### STEPS TO REPRODUCE

```
vagrant init ubuntu/trusty64
vagrant up
vagrant ssh

root@test1:~# ansible -i "localhost" -m setup localhost|grep ansible_service_mgr
        "ansible_service_mgr": "service",
```

##### EXPECTED RESULTS

```
 "ansible_service_mgr": "upstart"
```

**Comment by sivel** [comment]
Can you test with the 2.4.1rc2 or just even the stable-2.4 branch I believe this was fixed already.

**Comment by pescobar** [comment]
it doesn't seem to be fixed upstream

```
vagrant init ubuntu/trusty64
vagrant up
vagrant ssh

root@test1:/opt# git clone https://github.com/ansible/ansible.git --recursive
root@test1:/opt# source ansible/hacking/env-setup

root@test1:/opt# ansible --version
ansible 2.5.0 (devel ab8bacdf5e) last updated 2017/10/20 22:39:57 (GMT +000)

root@test1:/opt# ansible-config dump --only-changed
root@test1:/opt#

root@test1:/opt# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty

root@test1:/opt# ansible -i "localhost" -m setup localhost|grep ansible_service_mgr
"ansible_service_mgr": "service",
```

**Comment by noplanman** [comment]
Is there a fix for this yet?

Or do you have any smart quick-fix solution for it? (like setting the fact manually?)

**Comment by ashemedai** [comment]
Yes, the current devel and stable-2.4 branch has this since 6 days: https://github.com/ansible/ansible/commit/6203e899f03a923251e5cae72db3648aa4f02eee

@alikins Might want to close this ticket as well then. And I hope this leads to a 2.4.1.1 because it actively breaks existing deployments. Just ran into it myself today that playbooks stopped running.

**Comment by pescobar** [comment]
I just checked that this is fixed upstream

```
root@test1:~/ansible# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty

root@test1:~/ansible# ansible --version
ansible 2.5.0 (devel 26583adb58) last updated 2017/11/01 14:16:15 (GMT +000)

root@test1:~/ansible# ansible -i "localhost" -m setup localhost | grep ansible_service_mgr
        "ansible_service_mgr": "upstart",
```

**Comment by alikins** [comment]
Fixed in #32806 (and in stable-2.4 headed for 2.4.1)

**Comment by alikins** [comment]
> Fixed in #32806 (and in stable-2.4 headed for 2.4.1)

Er, make that 2.4.2
